Validation/RecoTau/src/TauValidationMiniAOD.cc

```diff
@@ -157,28 +536,205 @@ void TauValidationMiniAOD::analyze(const edm::Event& iEvent, const edm::EventSet
-      if (matchedTau->isTauIDAvailable("byDeepTau2017v2p1VSmuraw"))
-        byDeepTau2017v2p1VSmurawMap.find("")->second->Fill(matchedTau->tauID("byDeepTau2017v2p1VSmuraw"));
-      int j = 0;
-      for (const auto& it : discriminators_) {
+      puMap.find("")->second->Fill(pvHandle->size());
+      decayModeMap.find("pftau")->second->Fill(matchedTau->decayMode());
+
+      // fill select discriminators with matchedTau quantities
+      if (matchedTau->isTauIDAvailable("decayModeFindingNewDMs"))
+        decayModeFindingMap.find("")->second->Fill(matchedTau->tauID("decayModeFindingNewDMs"));
+      if (matchedTau->isTauIDAvailable("byDeepTau2018v2p5VSeraw"))
+        byDeepTau2018v2p5VSerawMap.find("")->second->Fill(matchedTau->tauID("byDeepTau2018v2p5VSeraw"));
+      if (matchedTau->isTauIDAvailable("byDeepTau2018v2p5VSjetraw"))
+        byDeepTau2018v2p5VSjetrawMap.find("")->second->Fill(matchedTau->tauID("byDeepTau2018v2p5VSjetraw"));
+      if (matchedTau->isTauIDAvailable("byDeepTau2018v2p5VSmuraw"))
+        byDeepTau2018v2p5VSmurawMap.find("")->second->Fill(matchedTau->tauID("byDeepTau2018v2p5VSmuraw"));
+
+      // fill tau mass for decay modes 0,1+2,5,6,7,10,11
+      if (matchedTau->decayMode() == 0) {
+        mtau_dm0Map.find("")->second->Fill(matchedTau->mass());
+        pTOverProng_dm0Map.find("")->second->Fill(matchedTau->pt(), matchedTau->ptLeadChargedCand());
+      } else if (matchedTau->decayMode() == 1 || matchedTau->decayMode() == 2) {
+        mtau_dm1p2Map.find("")->second->Fill(matchedTau->mass());
+        pTOverProng_dm1p2Map.find("")->second->Fill(matchedTau->pt(), matchedTau->ptLeadChargedCand());
+      } else if (matchedTau->decayMode() == 5) {
+        mtau_dm5Map.find("")->second->Fill(matchedTau->mass());
+        pTOverProng_dm5Map.find("")->second->Fill(matchedTau->pt(), matchedTau->ptLeadChargedCand());
+      } else if (matchedTau->decayMode() == 6) {
+        mtau_dm6Map.find("")->second->Fill(matchedTau->mass());
+        pTOverProng_dm6Map.find("")->second->Fill(matchedTau->pt(), matchedTau->ptLeadChargedCand());
+      } else if (matchedTau->decayMode() == 10) {
+        mtau_dm10Map.find("")->second->Fill(matchedTau->mass());
+        pTOverProng_dm10Map.find("")->second->Fill(matchedTau->pt(), matchedTau->ptLeadChargedCand());
+      } else if (matchedTau->decayMode() == 11) {
+        mtau_dm11Map.find("")->second->Fill(matchedTau->mass());
+        pTOverProng_dm11Map.find("")->second->Fill(matchedTau->pt(), matchedTau->ptLeadChargedCand());
+      }
+
+      // fill decay mode population plot
+      ntau_vs_dmMap.find("")->second->Fill(taus->size(), matchedTau->decayMode());
+
+      //Fill decay mode migration 2D histogragms
+      //First do a gen Matching
+      unsigned genindex = 0;
+      for (std::vector<reco::GenParticle>::const_iterator genParticle = genParticles->begin();
+           genParticle != genParticles->end();
+           genParticle++) {
```

Comment:
A range based for loop is more convenient here

Reply:
Done